PureDOOM.h

```diff
@@ -7612,35 +7612,22 @@ const unsigned char* doom_get_framebuffer(int channels)
     {
         return screen_buffer;
     }
-    else if (channels == 3)
+    else if (channels == 3 || channels == 4)
     {
         for (i = 0, len = SCREENWIDTH * SCREENHEIGHT; i < len; ++i)
         {
-            int k = i * 3;
+            int k = i * channels;
             int kpal = screen_buffer[i] * 3;
             final_screen_buffer[k + 0] = screen_palette[kpal + 0];
             final_screen_buffer[k + 1] = screen_palette[kpal + 1];
             final_screen_buffer[k + 2] = screen_palette[kpal + 2];
+
+            if (channels == 4)
```

Comment:
I repeat code because this line creates branching in a loop that should be as tight as possible.
Altho branch prediction will make this fast anyway, I rather keep every channel code separated. It's not like it's a lot of code anyway, to guarantee tightess loop.

Reply:
Sure, I'm not too worried about the speed for that. 

1) the compiler might optimize it out anyway. 

2) your library works with a smaller bitmap, so that part of processing can be slower.

I think it should stay small because it allows the user to change the size themselves and/or modify the smaller bitmap if they so choose. This comment is in response to one of your todo items.

Looking back at this code again, if we care about speed maybe it would be faster to use something like memcpy?

Reply:
This could also be fixed by using a macro I'd think. 

Eg. DOOM_SUPPORT_ALPHA or something

Then the logic is done in preprocessing only once.